docs/developer/architecture/user_customization_and_persistence.rst

```diff
@@ -0,0 +1,187 @@
+User Customization & Persistence
+================================
+
+Context
+-------
+
+.. todo::
+
+   This section will be handled by the requirements PR.
+
+
+Design Choices
+--------------
+
+:term:`ConfigMap` is chosen as a unified data access and storage media for user
+editable configurations in a MetalK8s cluster based on the above requirements
+for the following reasons:
+
+* Ability to support Update operations on ConfigMap's with CLI and UI easily
+  using our already existing python kubernetes module.
+* Guarantee of adaptability and ease of changing the design and implementation
+  in cases were customer needs evolve rapidly.
+* ConfigMaps are stored in the :term:`etcd` database which is generally being
+  backed up. This ensures that user settings cannot be lost easily.
+
+
+How it works
+~~~~~~~~~~~~
+
+Before Bootstrap, :file:`defaults.yaml` will be pre-filled with default service
+settings. Service pods and deployments will be configured to either
+consume configuration data directly from ConfigMaps or fallback to
+the default values if the service specific ConfigMap was not found.
+
+**Using Salt states**
+
+Once a ConfigMap is updated by the user (say a user changes the number of
+replicas for Prometheus deployments to a new value), then perform the
+following actions:
+
+  - Apply a Salt states that reads the ConfigMap object, validates the schema
+    and checks the new values passed.
+
+Storage format
+~~~~~~~~~~~~~~
+
+A sample ConfigMap can be defined with the following fields.
```

Comment:
Why this format? Explain in the design, compare with flat KV approach and why we prefer this one.

Reply:
Done